### ISSUE TITLE
fix(kvmmu): do not report unsupported cache eviction as successful

### ISSUE THREAD

**Issue by anthony-chaudhary** [anthony-chaudhary/fak#12622]
# fix(kvmmu): do not report an unsupported cache eviction as successful

```routing
lane: kvmmu
paths: ["internal/kvmmu/kvmmu.go", "internal/kvmmu/attention.go", "internal/kvmmu/audit_unsupported_evict_result_test.go"]
expected_steps: 4
```

Audited snapshot: `64cf527c1ca925822357ec113c6af37c2108084c`.

## Parent context

Correctness follow-up to closed #1704, which made KV-MMU consult `CanEvict` so hybrid recurrent caches skip unsupported physical eviction instead of panicking.

## Problem frame

- Centrality: Core
- P1 Context: preserved - public KV-MMU results must distinguish a physically removed span from one retained because its backend cannot evict.
- P2 Net value: advanced - callers stop advancing bookkeeping or declaring reclaimed capacity after a no-op.
- P3 Adaptation: preserved - hybrid recurrent caches continue using the existing safe skip behavior.
- P4 Operations: advanced - one internal result seam, its callers, and one deterministic backend-stub test.

## Current state

`Context.evict` returns zero immediately when an optional backend `CanEvict` rejects the operation (`kvmmu.go:288-290`). Several callers ignore that result and report success anyway:

- `AdmitResult` returns `evicted=true` after every quarantine verdict (`kvmmu.go:159-172`).
- `Quarantine` returns `ok=true` whenever the segment ID exists (`kvmmu.go:175-184`).
- `Compact` increments `swapped` for skipped removals before appending a summary (`kvmmu.go:224-252`).
- `ApplyPlan` increments its eviction count for skipped removals (`kvmmu.go:255-282`).
- `EvictColdest` appends an `EvictedSpan` with `Positions: 0` (`attention.go:157-193`).

The standalone witness uses an ordinary deterministic backend wrapper whose `CanEvict` returns an unsupported verdict. The cache remains at five positions, yet `AdmitResult` says `evicted=true`, `Quarantine` returns `(0,true)`, and `EvictColdest` reports the zero-position span as evicted on all three runs.

Closed #1704 intentionally established the skip behavior for served hybrid Gated-DeltaNet caches. It did not cover result/accounting honesty after that skip. Current repository search finds direct `AdmitResult` use in `cmd/unseedemo` and model-backed KV-MMU tests; no current production gateway call to this concrete `kvmmu.Context` was found, so classify the new result mismatch as P2 while preserving the historical served-hybrid reachability evidence.

## Why this is next

The panic fix made unsupported eviction a normal outcome, but the surrounding APIs still treat it as success. A small result-propagation fix prevents false cache state without implementing recurrent eviction.

## Core through-line

The internal eviction primitive returns both positions removed and whether removal occurred. Every API count, boolean, swap, and eviction record derives from that physical outcome.

## Working spine

Backend capability check -> attempted physical removal -> truthful outcome -> caller-specific bookkeeping only after success.

## Gold-plating boundary

Do not implement Gated-DeltaNet recurrent-state eviction, change quarantine verdict folding, return quarantined logits, or redesign the ABI backend interface. Keep the existing safe skip and fix only its result propagation.

## Suggested implementation steps

1. Change the private `Context.evict` helper to expose a success outcome in addition to the removed-position count.
2. Make `AdmitResult`, `Quarantine`, `Compact`, and `ApplyPlan` count or report only successful physical removals.
3. Make `EvictColdest` omit skipped candidates from its returned eviction records while terminating without a false reclaimed count.
4. Add the deterministic unsupported-backend regression below and retain #1704's no-panic test.

## Done condition

- [ ] `AdmitResult` returns `evicted=false` when `CanEvict` rejects removal; quarantine logits remain stripped as today.
- [ ] `Quarantine` returns `ok=false` for a found span that was not removed.
- [ ] `Compact` and `ApplyPlan` do not count skipped removals or mutate their ledgers as if removal succeeded.
- [ ] `EvictColdest` does not return zero-position entries as evicted, and the existing hybrid no-panic regression stays green.

## Definition of Done

The exact regression passes, all eviction-result callers derive success from physical removal, and the existing hybrid cache remains resident without a panic when eviction is unsupported.

## Acceptance gate

`go test ./internal/kvmmu -run '^(TestAuditHybridQuarantineDoesNotClaimSkippedEviction|TestHybridCacheSkipsEvictionInsteadOfPanicking)$' -count=1 -v` passes, followed by the full `internal/kvmmu` package test.

## Closure binding

The resolving commit cites this issue and carries `(fak kvmmu)`.

## Witness

Save this standalone file as `internal/kvmmu/audit_unsupported_evict_result_test.go`:

```go
package kvmmu_test

import (
	"context"
	"errors"
	"testing"

	"github.com/anthony-chaudhary/fak/internal/abi"
	"github.com/anthony-chaudhary/fak/internal/kvmmu"
	"github.com/anthony-chaudhary/fak/internal/model"
)

type auditQuarantineGate struct{}

func (auditQuarantineGate) Admit(context.Context, *abi.ToolCall, *abi.Result) abi.Verdict {
	return abi.Verdict{Kind: abi.VerdictQuarantine, By: "audit"}
}

type auditUnsupportedEvictBackend struct{ abi.KVBackend }

func (b *auditUnsupportedEvictBackend) CanEvict() error {
	return errors.New("audit backend cannot evict its recurrent state")
}

func TestAuditHybridQuarantineDoesNotClaimSkippedEviction(t *testing.T) {
	m := model.NewSynthetic(model.Config{
		ModelType: "llama", VocabSize: 16, HiddenSize: 16, IntermediateSize: 32,
		NumLayers: 1, NumHeads: 2, NumKVHeads: 2, HeadDim: 8,
		RMSNormEps: 1e-5, RopeTheta: 10000, EOSTokenID: -1,
	})
	inner, ok := model.KVBackend(m.NewSession())
	if !ok { t.Fatal("model.KVBackend: ok=false") }
	backend := &auditUnsupportedEvictBackend{KVBackend: inner}
	c := kvmmu.NewBackendWithGate(backend, auditQuarantineGate{})
	c.Append("sys", "system", []int{1, 2, 3})

	v, evicted, _ := c.AdmitResult(context.Background(), "result", "lookup", []int{4, 5}, []byte("ordinary cached result"))
	if v.Kind != abi.VerdictQuarantine {
		t.Fatalf("verdict=%v, want QUARANTINE", v.Kind)
	}
	if evicted {
		t.Errorf("AdmitResult evicted=true, but backend rejected eviction and cache still has %d positions", c.CacheLen())
	}
	if n, ok := c.Quarantine("result"); n != 0 || ok {
		t.Errorf("Quarantine returned (%d,%v), want (0,false) when backend rejects eviction", n, ok)
	}
	if got := c.EvictColdest(1); len(got) != 0 {
		t.Errorf("EvictColdest reported skipped eviction as %+v, want no evicted spans", got)
	}
}
```

## Verifiable Witness

The audit ran this exact standalone file through a Go overlay without changing public source. After saving it at the path above, run:

```text
$ go test ./internal/kvmmu -run '^TestAuditHybridQuarantineDoesNotClaimSkippedEviction$' -count=3 -v -timeout=30s
=== RUN   TestAuditHybridQuarantineDoesNotClaimSkippedEviction
    audit_hybrid_evict_result_test.go:44: AdmitResult evicted=true, but backend rejected eviction and cache still has 5 positions
    audit_hybrid_evict_result_test.go:47: Quarantine returned (0,true), want (0,false) when backend rejects eviction
    audit_hybrid_evict_result_test.go:50: EvictColdest reported skipped eviction as [{ID:result EMA:0 Positions:0}], want no evicted spans
--- FAIL: TestAuditHybridQuarantineDoesNotClaimSkippedEviction (0.00s)
[same failures on all 3 runs]
FAIL github.com/anthony-chaudhary/fak/internal/kvmmu
```

## Likely files

- `internal/kvmmu/kvmmu.go`
- `internal/kvmmu/attention.go`
- `internal/kvmmu/audit_unsupported_evict_result_test.go`

Exact seams at the audited SHA:

- `internal/kvmmu/kvmmu.go:159-172` (`AdmitResult`)
- `internal/kvmmu/kvmmu.go:175-184` (`Quarantine`)
- `internal/kvmmu/kvmmu.go:224-252` (`Compact`)
- `internal/kvmmu/kvmmu.go:255-290` (`ApplyPlan`, `evict`)
- `internal/kvmmu/attention.go:157-193` (`EvictColdest`)

## Lane

kvmmu

## Expected steps

4

## Blast radius / fallback

Affected: any injected or model-backed KV backend that implements `CanEvict` and rejects removal, including hybrid recurrent caches. Unaffected: ordinary softmax KV backends whose `Evict` removes the requested positions, verdict selection, and quarantine output stripping. The safe caller fallback is to verify the returned removed-position count instead of trusting the boolean or list entry.

## Dedupe

All-state GitHub searches for `kvmmu skipped eviction reports success`, `Quarantine returns true CanEvict`, `EvictColdest Positions 0`, and `hybrid eviction result honesty` found no exact issue. Closed #1704 owns the hybrid panic/skip fix and is the parent context; it does not require truthful public results after a skipped eviction.
